consensus/istanbul/backend.go

```diff
@@ -97,9 +99,67 @@ type Backend interface {
 	// ParentBlockValidators returns the validator set of the given proposal's parent block
 	ParentBlockValidators(proposal Proposal) ValidatorSet
 
-	// RefreshValPeers will connect with all the validators in the validator connection set and disconnect validator peers that are not in the set
-	RefreshValPeers() error
-
 	// Authorize injects a private key into the consensus engine.
 	Authorize(address common.Address, publicKey *ecdsa.PublicKey, decryptFn DecryptFn, signFn SignerFn, signBLSFn BLSSignerFn)
 }
+
+// BackendForProxy provides the Istanbul backend application specific functions for Istanbul proxy
+type BackendForProxy interface {
+	// Address returns the owner's address
+	Address() common.Address
+
+	// IsProxiedValidator returns true if this node is a proxied validator
+	IsProxiedValidator() bool
+
+	// IsProxy returns true if this node is a proxy
+	IsProxy() bool
+
+	// SelfNode returns the owner's node (if this is a proxy, it will return the external node)
+	SelfNode() *enode.Node
+
+	// Sign signs input data with the backend's private key
+	Sign([]byte) ([]byte, error)
```

Comment:
Could you clarify what key this is exactly? In particular, if this method is called on a Proxy node, what happens? Does it sign with its node key, or simply return an error?

Reply:
It's the validator's signing ecdsa key, and I commented as such.